GeoLib/PointVec.cpp

```diff
@@ -33,23 +33,51 @@ PointVec::PointVec (const std::string& name, std::vector<Point*>* points,
 	TemplateVec<Point> (name, points, name_id_map),
 	_type(type),
 	_aabb(points->begin(), points->end()),
-	_rel_eps(rel_eps)
+	_rel_eps(rel_eps*sqrt(MathLib::sqrDist(_aabb.getMinPoint(), _aabb.getMaxPoint()))),
+	_oct_tree(OctTree<GeoLib::Point, 16>::createOctTree(
+		_aabb.getMinPoint(), _aabb.getMaxPoint(), _rel_eps))
 {
 	assert (_data_vec);
-	std::size_t const number_of_all_input_pnts (_data_vec->size());
+	std::size_t const number_of_all_input_pnts(_data_vec->size());
 
-	_rel_eps *= sqrt(MathLib::sqrDist (_aabb.getMinPoint(),_aabb.getMaxPoint()));
-	makePntsUnique (_data_vec, _pnt_id_map, _rel_eps);
+	// correct the ids if necessary
+	for (std::size_t k(0); k<_data_vec->size(); ++k) {
+		if ((*_data_vec)[k]->getID() == std::numeric_limits<std::size_t>::max()) {
+			(*_data_vec)[k]->setID(k);
```

Comment:
it looks you assume a point ID should be an index in `_data_vec`. then it might be safer if you check `if ((*_data_vec)[k]->getID() == k)`? otherwise, there could be points having the same ID